### PR TITLE
Copy number integration and sklearn version update

### PR DESCRIPTION
**Pull request by gwaygenomics**
Major update integrating copy number data, updating pipeline for appropriate cross validation grid search, and adding logic to view cross validation test set performance.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,20 +4,41 @@
...
-We have previously described the ability of a machine learning classifier to detect an
-NF1 inactivation signature using Glioblastoma data. We applied an ensemble of logistic
-regression classifiers to the problem, but the solutions were unstable and overfit. To
-address these issues, we posited that we could leverage data from diverse tissue-types
-to build a pancancer NF1 classifier. We also predicted that an NF1/RAS classifier would
-better key in on NF1 inactivation since NF1 directly inhibits RAS activity and there
-are many more examples of samples with RAS mutations.
+We have previously described the ability of a machine learning classifier to
+detect an NF1 inactivation signature using Glioblastoma data
+([Way _et al._ 2016](http://dx.doi.org/10.1101/075382)). We applied an ensemble
```
Comment (dhimmel) [inline README.md]: Trivial but I prefer `https://doi.org/10.1101/075382`

File: README.md [hunk 2]
```diff
@@ -4,20 +4,41 @@
...
+
+We also provide the ability to build classifiers using publicly available
+data retrieved from [UCSC Xena](xena.ucsc.ecu).
+
+## Cancer Genes
+
+Note that in order to use the copy number integration feature, an additional
+file must be downloaded. The file is `Supplementary Table S2` of
+[Vogelstein _et al._ 2013]("http://doi.org/10.1126/science.1235122"). It can
+be downloaded [here]("science.sciencemag.org/highwire/filestream/594203/field_highwire_adjunct_files/1/1235122TablesS1-4.xlsx").
```
Comment (dhimmel) [inline README.md]: Wouldn't it be better to script this? Even a one line shell command you put in the README here.

File: environment.yml [hunk 3]
```diff
@@ -3,9 +3,8 @@ dependencies:
 - jupyter=1.0.0
 - numpy=1.10.4
 - pandas=0.18.0
-- scikit-learn=0.17.1
+- scikit-learn=0.18
```
Comment (dhimmel) [inline environment.yml]: I may specify further `0.18.0`.

**Comment by dhimmel** [comment]
Regarding storing results, I don't think it necessarily needs to be in this PR, but should happen before any future PRs are created.

File: README.md (hunk 2 above)
Comment (gwaygenomics) [inline README.md]: yeah, good thinking - i will add to next commit

**Comment by gwaygenomics** [comment]
> Would it make sense for you to create a results directory perhaps? You wouldn't have to save any big files if you wanted to keep things light. But perhaps some basic summary stats like what the model parameters are and what the performance is.

With several deadlines, the code base and analyses are moving very quickly - the pull request process is having a difficult time keeping up. Additionally, to comply with all data embargo guidelines, I cannot share any result made with this data.

> Regarding storing results, I don't think it necessarily needs to be in this PR, but should happen before any future PRs are created.

By storing results, you mean [this](https://github.com/greenelab/pancancer/pull/3/files#diff-be2ec6781838a325472989b583c52f24R569)?

> One thing I've noticed is that it's not really realistic for a reviewer to be able to assess entire changeset here. There's just too much changing. It would really help to see output and the diff of output from previous runs. 

I apologize for the large changes - although i don't really think this pull request is changing _that_ much. If it helps, I can annotate specific changes to the code using github comments. Really, there are three main changes.

1. Updating to sklearn 18.0.0
2. copy number integration
3. cross validation eval

**Comment by dhimmel** [comment]
> I apologize for the large changes - although i don't really think this pull request is changing that much.

It's not really an issue of how big the pull request is. It's that the script is big and that there's no tracked output to see the effect of the code. Just something to keep in mind for the future.

> By storing results, you mean [this](https://github.com/greenelab/pancancer/pull/3/files#diff-be2ec6781838a325472989b583c52f24R569)?

Exactly, summary results like that are very useful when tracked. Then the diff tells you the functional effect of your changes.

> Additionally, to comply with all data embargo guidelines, I cannot share any result made with this data.

Yep I understand. With #OAWeek2016 kicking off today and since this discussion is public, I'll take the opportunity to note how the embargo makes analyses difficult to review. Things that are difficult to review develop deficits.

@gwaygenomics -- Preemptive approval: merge at will once you address the minor comments.

**Pull request merged**